aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -960,22 +992,27 @@ Tensor full(
     Scalar fill_value,
     optional<DimnameList> names,
     const TensorOptions& options) {
-  auto result = at::empty(size, names, options);
+
+  if (options.layout() == kSparse) {
+    AT_ERROR("full(...) is not implemented for sparse layout");
+  }
+
+  auto result = at::empty(size, names, infer_full_options(fill_value, options));
```

Comment:
what was happening here before?  Does fill...fill the spare tensor values?

Reply:
You'll get an uninformative runtime error. 

There's actually a few things going on here. The standard full already had this check (https://github.com/pytorch/pytorch/blob/7dee36a0613f05795a935bc8deafdc0a8f6612b1/aten/src/ATen/native/TensorFactories.cpp#L346). This code path happens when explicitly specifying the `names` kwarg, and when a user specifies names and requests a sparse tensor from full they received a runtime error that named tensors don't support the sparse layout. That runtime error is probably why this check didn't appear here previously. However, you can specify None for `names`:

```
torch.full((1,), 1, names=None, layout=torch.sparse_coo)
: RuntimeError: Cannot access data pointer of Tensor that doesn't have storage
```

And get that uninformative runtime error. So I thought I'd copy the check to improve the error message in this (narrow) case. I should probably update it to use TORCH_CHECK, now that we're looking at it.